python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -274,55 +279,49 @@ def agg(self, func):
         1  1.5  1.75  2.0   2.0
         2  3.0  3.00  1.0   1.0
         """
-        normalized_aggs = self._normalize_aggs(func)
+        column_names, columns, normalized_aggs = self._normalize_aggs(func)
 
         # Note: When there are no key columns, the below produces
         # a Float64Index, while Pandas returns an Int64Index
         # (GH: 6945)
-        result = cudf.DataFrame._from_data(
-            *self._groupby.aggregate(self.obj, normalized_aggs)
+        (
+            result_columns,
+            grouped_key_cols,
+            included_aggregations,
+        ) = self._groupby.aggregate(columns, normalized_aggs)
+
+        result_index = self.grouping.keys._from_columns_like_self(
+            grouped_key_cols,
         )
 
+        multilevel = _is_multi_agg(func)
+        data = {}
+        for col_name, aggs, cols in zip(
+            column_names, included_aggregations, result_columns
+        ):
+            data.update(
+                {
+                    (col_name, agg.__name__ if callable(agg) else agg)
+                    if multilevel
+                    else col_name: col
```

Comment:
It's a little difficult to parse this and realize that the conditional is just for the dict key. An extra pair of parentheses might help.